examples/nas/enas/retrain.py

```diff
@@ -0,0 +1,435 @@
+
+            acc = accuracy(logits, y)
+            loss = criterion(logits, y)
+            loss = loss + aux_weight * aux_loss
+
+            optimizer.zero_grad()
+            loss.backward()
+            grad_norm = 0
+            trainable_params = child_model.parameters()
+
+            for param in trainable_params:
+                grad_norm = nn.utils.clip_grad_norm_(param, child_grad_bound)  # clip grad
+                param = grad_norm
+
+            optimizer.step()
+
+            if step % log_every == 0:
+                curr_time = time.time()
+                log_string = ""
+                log_string += "epoch={:<6d}".format(epoch)
+                log_string += "ch_step={:<6d}".format(step)
+                log_string += " loss={:<8.6f}".format(loss)
+                log_string += " lr={:<8.4f}".format(lr)
+                log_string += " |g|={:<8.4f}".format(grad_norm)
+                log_string += " tr_acc={:<8.4f}/{:>3d}".format(acc['acc1'], logits.size()[0])
+                log_string += " mins={:<10.2f}".format(float(curr_time - start_time) / 60)
+                logger.info(log_string)
+
+        epoch += 1
+        save_state = {
+            'step': step,
+            'epoch': epoch,
+            'child_model_state_dict': child_model.state_dict(),
+            'optimizer_state_dict': optimizer.state_dict()}
+        #         print(' Epoch {:<3d} loss: {:<.2f} '.format(epoch, loss))
+        torch.save(save_state, model_save_path)
+        child_model.eval()
+
+        if epoch % FLAGS.eval_every_epochs ==0:
+            logger.info("Epoch {}: Eval".format(epoch))
+            eval_acc, eval_loss = eval_once(child_model, device, "test", criterion, test_dataloader=test_dataloader)
+            logger.info(
+                "ch_step={} {}_accuracy={:<6.4f} {}_loss={:<6.4f}".format(step, "test", eval_acc, "test", eval_loss))
+            if eval_acc > best_acc:
+                best_acc = eval_acc
+                logger.info("Save best model")
+                save_state = {
+                    'step': step,
+                    'epoch': epoch,
+                    'child_model_state_dict': child_model.state_dict(),
+                    'optimizer_state_dict': optimizer.state_dict()}
+            torch.save(save_state, best_model_save_path)
+
+            result['accuracy'].append('Epoch {} acc: {:<6.4f}'.format(epoch, eval_acc,))
+            acc_l.append(eval_acc)
+
+        print(result['accuracy'][-1])
+
+    print('max acc %.3f at epoch: %i'%(max(acc_l), np.argmax(np.array(acc_l))))
+    print('Time cost: %.4f hours'%( float(time.time() - start_time) /3600.  ))
+    return result
+
+aux_weight = 0.4
+result = {'accuracy':[],}
+def dump_global_result(res_path,global_result, sort_keys = False):
+    with open(res_path, "w") as ss_file:
+        json.dump(global_result, ss_file, sort_keys=sort_keys, indent=2)
+
+def main():
+    parse_args()
+
+    os.environ['CUDA_VISIBLE_DEVICES'] = '4'
```

Comment:
Please remove this. This should be done in bash script.

Reply:
Sure. Could you please tell me how to?

Reply:
For example, `CUDA_VISIBLE_DEVICES=4 python retrain.py`